hawkular-datamining-api/src/main/java/org/hawkular/datamining/api/base/InMemorySubscriptionManager.java

```diff
@@ -86,33 +80,35 @@ public void subscribe(Subscription subscription) {
         }
 
         /**
+         * Set prediction listener
+         */
+        subscription.forecaster().setPredictionListener(predictionListener);
+
+        /**
          * Initialize subscription with old data
          */
         List<DataPoint> points = restMetricsClient.loadPoints(subscription.getMetric().getMetricId(),
                 subscription.getMetric().getTenant());
         subscription.forecaster().learn(points);
 
-        /**
-         * Set prediction listener
-         */
-        subscription.forecaster().setPredictionListener(predictionListener);
-
-        Logger.LOGGER.subscribing(subscription.getMetric().getMetricId(), subscription.getMetric().getTenant());
         tenantSubscriptions.getSubscriptions().put(subscription.getMetric().getMetricId(), subscription);
+        Logger.LOGGER.subscribe(subscription.getMetric().getMetricId(), subscription.getMetric().getTenant());
     }
 
     @Override
-    public void unSubscribeAll(String tenant, String metricId) {
-        unSubscribe(tenant, metricId, Subscription.SubscriptionOwner.getAllDefined());
+    public void unsubscribeAll(String tenant, String metricId) {
+        unsubscribe(tenant, metricId, Subscription.SubscriptionOwner.getAllDefined());
     }
 
     @Override
-    public void unSubscribe(String tenant, String metricId, Subscription.SubscriptionOwner subscriptionOwner) {
-        unSubscribe(tenant, metricId, new HashSet<>(Arrays.asList(subscriptionOwner)));
+    public void unsubscribe(String tenant, String metricId, Subscription.SubscriptionOwner subscriptionOwner) {
+        unsubscribe(tenant, metricId, new HashSet<>(Arrays.asList(subscriptionOwner)));
     }
 
     @Override
-    public void unSubscribe(String tenant, String metricId, Set<Subscription.SubscriptionOwner> subscriptionOwners) {
+    public synchronized void unsubscribe(String tenant, String metricId,
```

Comment:
Again, as with `subscribe` I'd synchronize on `subscriptions` rather than on `this`.